filebeat/harvester/registry.go

```diff
@@ -39,9 +45,7 @@ func (r *Registry) Stop() {
 	close(r.done)
 
 	for _, hv := range r.harvesters {
-		r.wg.Add(1)
 		go func(h Harvester) {
-			r.wg.Done()
 			h.Stop()
 		}(hv)
 	}
```

Comment:
Thinking again about the above: In my implementation, because of the lock Start can only be called again after it is guaranteed that the channel is closed, means no new harvesters will be started.